### ISSUE TITLE
New Rule: Telegram Bot Token (@secretlint/secretlint-rule-telegram)

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#1532]
## Description

Telegram Bot Tokens have a distinctive format that makes them uniquely identifiable with very low false positive risk. GitHub Secret Scanning supports these as Push Protection targets.

Leaked Telegram bot tokens allow full control of the bot, including reading messages and sending messages on behalf of the bot.

## Confidence

The NG samples in this issue were detected by **GitHub Secret Scanning as real "Public leak" alerts** immediately after this issue was published. This confirms:

- The proposed regex matches the same pattern GitHub uses for Push Protection
- Real Telegram Bot Tokens would be reliably detected
- High-confidence rule candidate with zero expected false positives

## Official Source

- Telegram Bot API documentation: https://core.telegram.org/bots/api#authorizing-your-bot
- GitHub Push Protection: supported (listed under "Telegram" provider)
- gitleaks: https://github.com/gitleaks/gitleaks (has detection rule)

Telegram's documentation shows the token format as `123456:ABC-DEF1234ghIkl-zyx57W2v1u123ew11`. The structure (numeric bot ID + `:A` + 34 characters) is empirically consistent across all known tokens.

### Bot Token

- **Message ID:** `TELEGRAM_BOT_TOKEN`
- **Prefix:** `{5-16 digit bot ID}:A`
- **Character set after `:A`:** `[A-Za-z0-9_-]`
- **Length after `:A`:** 34 characters
- **Regex:**

```regex
/(?<!\p{L})[0-9]{5,16}:A[A-Za-z0-9_\-]{34}(?!\p{L})/gu
```

## OK Examples (should NOT be detected)

```
123:Aabcdef
1234:Ashort
some_random_text:notabot
```

## NG Examples (should be detected)

> Note: the example values below are dummies (verified by GitHub Secret Scanning as matching the production pattern).

```
123456789:AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
1000000000:ABBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB
9999999999:ACCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC
```
